### PR TITLE
feat: Header Components - tab through navigation using keyboard

### PR DESCRIPTION
**Pull request by martind-bjss**
<details>
  <summary>📦 Published PR as canary version: <code>0.3.10--canary.62.dbf76a9.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @ukho/admiralty-angular@0.3.10--canary.62.dbf76a9.0
  npm install @ukho/admiralty-core@0.3.10--canary.62.dbf76a9.0
  # or 
  yarn add @ukho/admiralty-angular@0.3.10--canary.62.dbf76a9.0
  yarn add @ukho/admiralty-core@0.3.10--canary.62.dbf76a9.0
  ```
</details>

### CODE REVIEW

File: packages/core/src/components/header-menu-item/header-menu-item.scss [hunk 1]
```diff
@@ -74,9 +96,9 @@
         background-color: colours.$colour-nav-bg-hover;
       }
 
-      &:focus {
-        box-shadow: colours.$colour-utility-focus inset 0 0 0 3px;
-      }
+      // &:focus {
+      //   box-shadow: colours.$colour-utility-focus inset 0 0 0 3px;
+      // }
```
Comment (UKHOGreg) [inline packages/core/src/components/header-menu-item/header-menu-item.scss:101]: comment should be removed?

File: packages/core/src/components/header-menu-item/header-menu-item.tsx [hunk 2]
```diff
@@ -43,41 +43,79 @@ export class HeaderMenuItemComponent {
...
+  }
+
+  private closeSubMenu() {
+    const subMenu: HTMLDivElement = this.el.querySelector('div.sub-menu');
+    subMenu.classList.add('desktop-hide');
+  }
+
+  handleClick(ev: MouseEvent): CustomEvent<void> {
     ev.stopPropagation();
+    console.log('clicked!');
```
Comment (UKHOGreg) [inline packages/core/src/components/header-menu-item/header-menu-item.tsx:66]: remove log?

File: packages/core/src/components/header-menu-item/header-menu-item.tsx [hunk 3]
```diff
@@ -43,41 +43,79 @@ export class HeaderMenuItemComponent {
     }
   }
 
-  handleClick = (ev: MouseEvent): CustomEvent<void> => {
+  private toggleSubMenu() {
+    const subMenu: HTMLDivElement = this.el.querySelector('div.sub-menu');
+    console.log('clicked-subMenuExists, display:', subMenu.style.display);
```
Comment (UKHOGreg) [inline packages/core/src/components/header-menu-item/header-menu-item.tsx:48]: remove log?

File: packages/core/src/components/header-menu-item/header-menu-item.tsx [hunk 4]
```diff
@@ -43,41 +43,79 @@ export class HeaderMenuItemComponent {
...
     if (this.subMenuExists) {
       this.subMenuSelector.setAttribute('aria-hidden', 'false');
+
+      this.toggleSubMenu();
     }
-  };
+  }
 
   /**
    * Handles mouseover events, used to set the aria hidden attribute when submenu
```
Comment (UKHOGreg) [inline packages/core/src/components/header-menu-item/header-menu-item.tsx:89]: mouseout event, right?

**Review by UKHOGreg — CHANGES_REQUESTED** [review]

**Comment by UKHOGreg** [comment]
Header with no title still focuses on the empty title

image

**Comment by ecstaticrainbow** [comment]
I think we should not render the title html if the title prop is falsey, just to avoid unnecessary tab stops

**Review by ecstaticrainbow — APPROVED** [review]
LGTM

**Comment by UKHOGreg** [comment]
UI review on TB's behalf

**Pull request merged**